flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/TemporalJoinITCase.scala

```diff
@@ -158,11 +160,87 @@ class TemporalJoinITCase(state: StateBackendMode)
 
     assertEquals(expectedOutput, sink.getAppendResults.toSet)
   }
+
+  @Test
+  def testNestedTemporalJoin(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = StreamTableEnvironment.create(env, TableTestUtil.STREAM_SETTING)
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+
+    val sqlQuery =
+      """
+        |SELECT
+        |  o.amount, r.rate, p.price
```

Comment:
If we are using a real world example, could we make it more useful? 

How about:
```
Orders
  |- orderId: String
  |- productId: String
  |- amount: Long
  |- timestamp: Timestamp

Rates
  |- currency: String
  |- rate: Double
  |- timestamp: Timestamp

Prices
  |- productId: String
  |- currency: String
  |- price: Double
  |- timestamp: Timestamp

Query:
SELECT
  o.orderId,
  (o.amount * p.price * r.rate) as total_price
FROM
  Orders AS o,
  LATERAL TABLE (Rates(o.rowtime)) AS r,
  LATERAL TABLE (Prices(o.rowtime)) AS p
WHERE
  o.productId = p.productId AND 
  r.currency = p.currency
```